core/app/models/concerns/spree/product_scopes.rb

```diff
@@ -0,0 +1,254 @@
+module Spree
+  module ProductScopes
+    extend ActiveSupport::Concern
+
+    included do
+      cattr_accessor :search_scopes do
+        []
+      end
+
+      def self.add_search_scope(name, &block)
+        singleton_class.send(:define_method, name.to_sym, &block)
+        search_scopes << name.to_sym
+      end
+
+      def self.simple_scopes
+        [
+          :ascend_by_updated_at,
+          :descend_by_updated_at,
+          :ascend_by_name,
+          :descend_by_name
+        ]
+      end
+
+      def self.add_simple_scopes(scopes)
+        scopes.each do |name|
+          # We should not define price scopes here, as they require something slightly different
+          next if name.to_s.include?('master_price')
+
+          parts = name.to_s.match(/(.*)_by_(.*)/)
+          scope(name.to_s, -> { order(Arel.sql("#{Product.quoted_table_name}.#{parts[2]} #{parts[1] == 'ascend' ? 'ASC' : 'DESC'}")) })
+        end
+      end
+
+      def self.property_conditions(property)
+        properties = Property.table_name
+        case property
+        when String   then { "#{properties}.name" => property }
+        when Property then { "#{properties}.id" => property.id }
+        else { "#{properties}.id" => property.to_i }
+        end
+      end
+
+      add_simple_scopes simple_scopes
```

Comment:
Why did you put this code `add_simple_scopes simple_scopes` here?

Reply:
@PSzyszkaSpark same here :)